examples/chaincode/go/chaincode_nondeter_example02.go

```diff
@@ -0,0 +1,198 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/openblockchain/obc-peer/openchain/chaincode/shim"
+        "math/random"
+
+)
+
+// SimpleChaincode example simple Chaincode implementation
+type SimpleChaincode struct {
+}
+
+func (t *SimpleChaincode) init(stub *shim.ChaincodeStub, args []string) ([]byte, error) {
+	var A, B string    // Entities
+	var Aval, Bval int // Asset holdings
+	var err error
+
+	if len(args) != 4 {
+		return nil, errors.New("Incorrect number of arguments. Expecting 4")
+	}
+
+	// Initialize the chaincode
+	A = args[0]
+	Aval, err = strconv.Atoi(args[1])
+	if err != nil {
+		return nil, errors.New("Expecting integer value for asset holding")
+	}
+	B = rand.Intn(100)
```

Comment:
I didn't a seed() function being called which means that rand.Intn() will be deterministic and always return the same sequence of numbers across all the peers.

I think that every time you want to generate a random number, you need to do something like rand.Seed(time now) then rand.Intn().  See this [example](https://gobyexample.com/random-numbers) 

Also, I think it will be better if you do this in the invoke() method  and use an random number for the X value rather than making the Bval random.